### PR TITLE
add VGF405X8

### PR DESCRIPTION
**Pull request by vgpengke**

### CODE REVIEW

**Comment by nerdCopter** [comment]
which one is good #226 or this #225?
you can fix/commit to your branch and push it again rather than opening multiple PR's.

**Comment by vgpengke** [comment]
> #226和#225哪一个好？ 您可以修复/提交您的分支并再次推送它，而不是打开多个 PR。
 Please  add #226, thank you！

File: configs/VGF405X8/config.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU     STM32F405
+
+#define BOARD_NAME        VDRCF4X8
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:26]: Does not correspond with the directory name.

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: configs/VGF405X8/config.h (hunk 1 above)
Comment (vgpengke) [reply configs/VGF405X8/config.h]: Name has been changed

File: configs/VGF405X8/config.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    TIMER_PIN_MAP( 4, PC8 , 2,  1) \
+    TIMER_PIN_MAP( 5, PC9 , 2,  0) \
+    TIMER_PIN_MAP( 6, PB0 , 2,  0) \
+    TIMER_PIN_MAP( 7, PB1 , 2,  0)
+
+
+#define ADC1_DMA_OPT        0
+
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define BARO_I2C_INSTANCE (I2CDEV_2)
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:98]: Don't see I2C pins defined for port 2

File: configs/VGF405X8/config.h [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+#define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 179
+#define BEEPER_INVERTED
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_INSTANCE SPI1
+#define FLASH_SPI_INSTANCE SPI2
+#define USE_SPI_GYRO
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:108]: already in target.h

File: configs/VGF405X8/config.h [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+#define GYRO_1_SPI_INSTANCE SPI3
+#define GYRO_1_ALIGN CW270_DEG
+#define GYRO_1_ALIGN_YAW 2700
+#define GYRO_2_SPI_INSTANCE SPI3
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:112]: Does this board have a second gyro ?

File: configs/VGF405X8/config.h [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    TIMER_PIN_MAP( 4, PC8 , 2,  1) \
+    TIMER_PIN_MAP( 5, PC9 , 2,  0) \
+    TIMER_PIN_MAP( 6, PB0 , 2,  0) \
+    TIMER_PIN_MAP( 7, PB1 , 2,  0)
+
+
+#define ADC1_DMA_OPT        0
+
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:98]:
```suggestion
#define BARO_I2C_INSTANCE (I2CDEV_1)
```

File: configs/VGF405X8/config.h [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+#define DEFAULT_BLACKBOX_DEVICE     BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 179
+#define BEEPER_INVERTED
+#define SYSTEM_HSE_MHZ 8
+
```
Comment (haslinghuis) [inline configs/VGF405X8/config.h:106]: Also needs these retained:
```suggestion
#define MAX7456_SPI_INSTANCE SPI1
#define FLASH_SPI_INSTANCE SPI2
```

**Comment by haslinghuis** [comment]
Just notice this PR is based on master instead of a future branch.

image

**Comment by vgpengke** [comment]
> 请注意，此 PR 是基于 master 而不是未来的分支。
> 
> 图像

**Pull request closed without merging**